tests/functional/agent/agent_functional_test.py

```diff
@@ -94,6 +94,30 @@ def test_end2end(run_input_map, delete_agents_and_team_agents):
     agent.delete()
 
 
+def test_python_interpreter_tool(delete_agents_and_team_agents):
```

Comment:
We should create a functional test for running with custom code as well, if it's not too costly.